static/js/stream_user_group_access_data.js

```diff
@@ -0,0 +1,91 @@
+import _ from "lodash";
+// This stores stream_user_group_access objects mapped according to their id.
+let access_objs_by_id_map;
+
+// This stores all the stream_user_group_access objects of a stream as a array
+// with key as stream_id.
+let access_objs_by_stream_map;
+export function init() {
+    access_objs_by_id_map = new Map();
+    access_objs_by_stream_map = new Map();
+}
+// init function helps to clear data in tests.
+init();
+
+const stream_user_group_access_fields = ["id", "stream_id", "group_id"];
+
+export function add_access_obj(access_obj) {
+    const cleaned_obj = _.pick(access_obj, stream_user_group_access_fields);
+    access_objs_by_id_map.set(cleaned_obj.id, cleaned_obj);
+    add_to_stream_map(access_obj);
+}
+
+export function add_to_stream_map(access_obj) {
+    // This should be called from add_access_obj only so that data in both maps is
+    // always in sync.
+
+    // Adds new stream user_group_access obj in array of a stream only if it is
+    // is already not present
+    const cleaned_obj = _.pick(access_obj, stream_user_group_access_fields);
```

Comment:
Isn't the data passed to `add_to_stream_map` by `add_access_obj` already cleaned? Can't we skip this step and just call the argument of `add_to_stream_map` as `cleaned_access_obj` or something like that?